pkg/controller/seed-controller-manager/mla/alertmanager_controller.go

```diff
@@ -348,10 +348,13 @@ func (r *alertmanagerController) ensureAlertmanagerConfiguration(ctx context.Con
 	if err != nil {
 		return err
 	}
+
 	expectedConfig := map[string]interface{}{}
-	if err := yaml.UnmarshalStrict(config, &expectedConfig); err != nil {
+	decoder := yaml.NewDecoder(bytes.NewReader(config))
```

Comment:
missing `decoder.KnownFields(true)`?

Reply:
Maybe a func like this can protect against this and also be easy to grep. 

```
func NewStrictDecoder(r io.Reader) *yaml.Decoder {
	decoder := yaml.NewDecoder(r)
	decoder.KnownFields(true)
	return decoder
}

```